packages/@aws-cdk/aws-ecs-patterns/test/ec2/l3s-v2.test.ts

```diff
@@ -0,0 +1,1481 @@
+        {
+          ContainerName: 'myContainer',
+          ContainerPort: 90,
+          TargetGroupArn: {
+            Ref: 'ServicelblistenerECSTargetGroupmyContainer90GroupF5A6D3A0',
+          },
+        },
+      ],
+      PropagateTags: 'SERVICE',
+      ServiceName: 'myService',
+    });
+
+    expect(stack).toHaveResourceLike('AWS::ECS::TaskDefinition', {
+      ContainerDefinitions: [
+        {
+          Cpu: 256,
+          Environment: [
+            {
+              Name: 'TEST_ENVIRONMENT_VARIABLE1',
+              Value: 'test environment variable 1 value',
+            },
+            {
+              Name: 'TEST_ENVIRONMENT_VARIABLE2',
+              Value: 'test environment variable 2 value',
+            },
+          ],
+          Essential: true,
+          Image: 'test',
+          LogConfiguration: {
+            LogDriver: 'awslogs',
+            Options: {
+              'awslogs-group': {
+                Ref: 'ServiceTaskDefmyContainerLogGroup0A87368B',
+              },
+              'awslogs-stream-prefix': 'TestStream',
+              'awslogs-region': {
+                Ref: 'AWS::Region',
+              },
+            },
+          },
+          Memory: 1024,
+          MemoryReservation: 1024,
+          Name: 'myContainer',
+          PortMappings: [
+            {
+              ContainerPort: 80,
+              HostPort: 0,
+              Protocol: 'tcp',
+            },
+            {
+              ContainerPort: 90,
+              HostPort: 0,
+              Protocol: 'tcp',
+            },
+          ],
+          DockerLabels: {
+            label1: 'labelValue1',
+            label2: 'labelValue2',
+          },
+        },
+      ],
+      ExecutionRoleArn: {
+        'Fn::GetAtt': [
+          'ExecutionRole605A040B',
+          'Arn',
+        ],
+      },
+      Family: 'ServiceTaskDef79D79521',
+      NetworkMode: 'bridge',
+      RequiresCompatibilities: [
+        'EC2',
+      ],
+      TaskRoleArn: {
+        'Fn::GetAtt': [
+          'TaskRole30FC0FBB',
+          'Arn',
+        ],
+      },
+    });
+
+    expect(stack).toHaveResourceLike('AWS::ElasticLoadBalancingV2::Listener', {
+      Port: 443,
+      Protocol: 'HTTPS',
+      Certificates: [{
+        CertificateArn: 'helloworld',
+      }],
+      SslPolicy: SslPolicy.TLS12_EXT,
+    });
+  });
+
+  test('set vpc instead of cluster', () => {
+    // GIVEN
+    const stack = new Stack();
+    const vpc = new Vpc(stack, 'VPC');
+
+    // WHEN
+    new ApplicationMultipleTargetGroupsEc2Service(stack, 'Service', {
+      vpc,
+      memoryLimitMiB: 1024,
+      taskImageOptions: {
+        image: ContainerImage.fromRegistry('test'),
+      },
+    });
+
+    // THEN - stack does not contain a LaunchConfiguration
+    const template = SynthUtils.synthesize(stack, { skipValidation: true });
+    expect(template).not.toHaveResource('AWS::AutoScaling::LaunchConfiguration');
+    expect(() => SynthUtils.synthesize(stack)).toThrow();
```

Comment:
@nija-at I'd like a second pair of eyes on this test (there are 2 other tests with the same pattern as well). It is comparable to this test in nodeunit:

https://github.com/aws/aws-cdk/blob/c6c594159c7fbda66f40fe8666f70b6806bb2d5e/packages/%40aws-cdk/aws-ecs-patterns/test/ec2/test.l3s-v2.ts#L270-L273

The issue in question is the use of `expect(stack, true)` as the `true` refers to setting `skipValidation=true`. Without it, the test returns a validation error and we cannot check if the stack contains LaunchConfiguration or not. Since the Jest `expect()` function does not allow this, I dove in and found `SynthUtils.synthesize` exposes such an option.

I'm not sure if this is the best way to go about this so I wanted to bring it to your attention.

Reply:
Hey Kaizen. This is fine for now.

When we migrate this module to 'assertions', we will need to make the validation succeed.